sdk/core/azure-core/inc/azure/core/base64.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Utility functions to help convert between binary data and UTF-8 encoded text that is
+ * represented in base 64.
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief Encodes the vector of binary data into UTF-8 encoded text represented as base 64.
+   *
+   * @param data The input vector that contains binary data that needs to be encoded.
+   * @return The UTF-8 encoded text in base 64.
+   */
+  std::string Base64Encode(const std::vector<uint8_t>& data);
+
+  /**
+   * @brief Decodes the UTF-8 encoded text represented as base 64 into binary data.
+   *
+   * @param text The input UTF-8 encoded text in base 64 that needs to be decoded.
+   * @return The decoded binary data.
+   */
+  std::vector<uint8_t> Base64Decode(const std::string& text);
```

Comment:
Would we want to use the same name for for the parameter? data vs text? 
Maybe input data? Or src?
Just a nit

Reply:
> Would we want to use the same name for for the parameter?

No, it was intentionally left as different since they represent different things.

One takes any binary data that you want to convert to a text string, and the other is the opposite.